app/models/course/discussion/post.rb

```diff
@@ -1,5 +1,11 @@
 class Course::Discussion::Post < ActiveRecord::Base
   acts_as_forest order: :created_at
 
+  after_initialize :set_topic, if: :new_record?
```

Comment:
It depends ? sometimes we use both, like: https://github.com/Coursemology/coursemology2/blob/master/app/models/course.rb#L10